crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -119,7 +119,14 @@ fn infer_definition_types_cycle_recovery<'db>(
     _cycle: &salsa::Cycle,
     input: Definition<'db>,
 ) -> TypeInference<'db> {
-    tracing::trace!("infer_definition_types_cycle_recovery");
+    let file = input.file(db);
+    let _span = tracing::trace_span!(
+        "infer_definition_types_cycle_recovery",
+        range = ?input.kind(db).target_range(),
+        file = %file.path(db)
+    )
+    .entered();
+
```

Comment:
This is similar to `infer_definition_types` and it allowed me to quickly understand the origin of the cycle and what was causing it.